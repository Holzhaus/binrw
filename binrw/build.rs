use std::env;

fn main() {
    println!("cargo:rerun-if-env-changed=BINTRACE_LINK_SEARCH");
    println!("cargo:rerun-if-env-changed=BINTRACE_LINK_LIB");
    println!("cargo:rerun-if-changed=build.rs");

    if cfg!(bintrace) {
        if let Ok(link_search) = env::var("BINTRACE_LINK_SEARCH") {
            println!("cargo:rustc-link-search={}", link_search)
        }

        if let Ok(link_lib) = env::var("BINTRACE_LINK_LIB") {
            println!("cargo:rustc-link-lib={}", link_lib);
        } else {
            println!("cargo:warning=No bintrace backend provided");
        }
    }
}
