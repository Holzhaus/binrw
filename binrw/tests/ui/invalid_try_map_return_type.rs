#![feature(generic_associated_types)]
use binrw::BinRead;

#[derive(BinRead)]
struct Foo {
    #[br(try_map = |_| 0)]
    a: i32,
}

fn main() {}
