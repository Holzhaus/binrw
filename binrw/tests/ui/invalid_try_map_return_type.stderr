error[E0308]: mismatched types
 --> $DIR/invalid_try_map_return_type.rs:6:24
  |
6 |     #[br(try_map = |_| 0)]
  |                        ^ expected enum `Result`, found integer
  |
  = note: expected enum `Result<i32, _>`
             found type `{integer}`
help: try using a variant of the expected enum
  |
6 |     #[br(try_map = |_| Ok(0))]
  |                        ^^^^^
6 |     #[br(try_map = |_| Err(0))]
  |                        ^^^^^^
